File: SRC.NET/BattleConfigData.vb
Option Strict Off
Option Explicit On
Friend Class BattleConfigData
	
	' Copyright (C) 1997-2012 Kei Sakamoto / Inui Tetsuyuki
	'Invalid_string_refer_to_original_code
	'Invalid_string_refer_to_original_code
	'Invalid_string_refer_to_original_code
	
	'Invalid_string_refer_to_original_code
	'Invalid_string_refer_to_original_code
	
	'蜷咲ｧｰ
	Public Name As String
	
	'Invalid_string_refer_to_original_code
	Public ConfigCalc As String
	
	'Invalid_string_refer_to_original_code
	'UPGRADE_NOTE: Class_Initialize は Class_Initialize_Renamed にアップグレードされました。 詳細については、'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="A9E4979A-37FA-4718-9994-97DD76ED70A7"' をクリックしてください。
	Private Sub Class_Initialize_Renamed()
		ConfigCalc = ""
	End Sub
	Public Sub New()
		MyBase.New()
		Class_Initialize_Renamed()
	End Sub
	
	'Invalid_string_refer_to_original_code
	'Invalid_string_refer_to_original_code
	Public Function Calculate() As Double
		Dim expr As String
		Dim morales As Integer
		
		expr = ConfigCalc
		
		'繧ｳ繝ｳ繝輔ぅ繧ｰ螟画焚繧呈怏蜉ｹ縺ｫ縺吶ｋ
		BCVariable.IsConfig = True
		
		'蠑上ｒ隧穂ｾ｡縺吶ｋ
		Calculate = GetValueAsDouble(expr)
		
		'繧ｳ繝ｳ繝輔ぅ繧ｰ螟画焚繧堤┌蜉ｹ縺ｫ縺吶ｋ
		BCVariable.IsConfig = False
	End Function
End Class

File: SRC.Sharp/SRCCore/Models/BattleConfigData.cs

namespace Project1
{
    internal class BattleConfigData
    {

        // Copyright (C) 1997-2012 Kei Sakamoto / Inui Tetsuyuki
        // 本プログラムはフリーソフトであり、無保証です。
        // 本プログラムはGNU General Public License(Ver.3またはそれ以降)が定める条件の下で
        // 再頒布または改変することができます。

        // バトルコンフィグデータのクラス
        // --- ダメージ計算、命中率算出など、バトルに関連するエリアスを定義します。

        // 名称
        public string Name;

        // 計算式
        public string ConfigCalc;

        // クラスの初期化
        // UPGRADE_NOTE: Class_Initialize は Class_Initialize_Renamed にアップグレードされました。 詳細については、'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="A9E4979A-37FA-4718-9994-97DD76ED70A7"' をクリックしてください。
        private void Class_Initialize_Renamed()
        {
            ConfigCalc = "";
        }

        public BattleConfigData() : base()
        {
            Class_Initialize_Renamed();
        }

        // バトルコンフィグデータに基づいた置換＆計算の実行
        // 実行前に使う可能性のある変数を事前に代入しておくこと
        public double Calculate()
        {
            double CalculateRet = default;
            string expr;
            int morales;
            expr = ConfigCalc;

            // コンフィグ変数を有効にする
            BCVariable.IsConfig = true;

            // 式を評価する
            CalculateRet = Expression.GetValueAsDouble(ref expr);

            // コンフィグ変数を無効にする
            BCVariable.IsConfig = false;
            return CalculateRet;
        }
    }
}